### DISCUSSION TITLE
API para rede cnpj

### DISCUSSION THREAD

**Discussion by ndr3wpss** [rictom/rede-cnpj#11] [General]
Olá, boa tarde! Gostaria de saber se existe alguma API para o rede CNPJ. 
Valeuuuuuu

**Comment by rictom** [comment]
Eu coloquei um teste na versão online da redecnpj. 
Para pegar os dados do cnpj e socios, use a url 
https://www.redecnpj.com.br/rede/api/cnpj/CNPJ
sendo CNPJ o cnpj da empresa pesquisada (só os 14 digitos)
Por exemplo, para consultar os dados do Banco do Brasil, teste
https://www.redecnpj.com.br/rede/api/cnpj/00000000000191

As consultas tem um limite de 1 por segundo e outros limites por minuto, hora e dia. Se precisar fazer muitas consultas, sugiro fazer a consulta diretamente na base cnpj tratada em sqlite em 
https://github.com/rictom/cnpj-sqlite#arquivo_sqlite

**Reply by ndr3wpss** [reply]
Vlw!

**Reply by ndr3wpss** [reply]
Opa, beleza?
Quantas camadas retornam utilizando a API?

**Comment by ndr3wpss** [comment]
Para baixar os dados públicos do Gov, você utiliza algum script ou seria algo manual?

**Comment by rictom** [comment]
Eu tenho usado este programa https://www.freedownloadmanager.org/pt/
que permite baixar blocos de cada arquivo para o download ficar mais rápido.
O site da receita era muito lento para baixar, agora está melhor.
Vc copia a lista dos arquivos (funcionou até marco/2023)
http://200.152.38.155/CNPJ/Cnaes.zip
http://200.152.38.155/CNPJ/Empresas0.zip
http://200.152.38.155/CNPJ/Empresas1.zip
http://200.152.38.155/CNPJ/Empresas2.zip
http://200.152.38.155/CNPJ/Empresas3.zip
http://200.152.38.155/CNPJ/Empresas4.zip
http://200.152.38.155/CNPJ/Empresas5.zip
http://200.152.38.155/CNPJ/Empresas6.zip
http://200.152.38.155/CNPJ/Empresas7.zip
http://200.152.38.155/CNPJ/Empresas8.zip
http://200.152.38.155/CNPJ/Empresas9.zip
http://200.152.38.155/CNPJ/Estabelecimentos0.zip
http://200.152.38.155/CNPJ/Estabelecimentos1.zip
http://200.152.38.155/CNPJ/Estabelecimentos2.zip
http://200.152.38.155/CNPJ/Estabelecimentos3.zip
http://200.152.38.155/CNPJ/Estabelecimentos4.zip
http://200.152.38.155/CNPJ/Estabelecimentos5.zip
http://200.152.38.155/CNPJ/Estabelecimentos6.zip
http://200.152.38.155/CNPJ/Estabelecimentos7.zip
http://200.152.38.155/CNPJ/Estabelecimentos8.zip
http://200.152.38.155/CNPJ/Estabelecimentos9.zip
http://200.152.38.155/CNPJ/Motivos.zip
http://200.152.38.155/CNPJ/Municipios.zip
http://200.152.38.155/CNPJ/Naturezas.zip
http://200.152.38.155/CNPJ/Paises.zip
http://200.152.38.155/CNPJ/Qualificacoes.zip
http://200.152.38.155/CNPJ/Simples.zip
http://200.152.38.155/CNPJ/Socios0.zip
http://200.152.38.155/CNPJ/Socios1.zip
http://200.152.38.155/CNPJ/Socios2.zip
http://200.152.38.155/CNPJ/Socios3.zip
http://200.152.38.155/CNPJ/Socios4.zip
http://200.152.38.155/CNPJ/Socios5.zip
http://200.152.38.155/CNPJ/Socios6.zip
http://200.152.38.155/CNPJ/Socios7.zip
http://200.152.38.155/CNPJ/Socios8.zip
http://200.152.38.155/CNPJ/Socios9.zip
e cola do download manager.
